articles/bottom-navigation-bar-in-android/index.md

```diff
@@ -0,0 +1,290 @@
+       android:label="@string/title_search"
+       tools:layout="@layout/fragment_search" />
+
+   <fragment
+       android:id="@+id/navigation_profile"
+       android:name="com.example.bottomnavigationbar.profile"
+       android:label="@string/title_profile"
+       tools:layout="@layout/fragment_profile" />
+
+</navigation>
+```
+
+### Step 3 – Create a Bottom Navigation View.
+
+In this step, we will add the Bottom Navigation View to our activity's resource file. Add a Bottom Navigation View and a NavHost In your XML Layout file.
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+
+<androidx.constraintlayout.widget.ConstraintLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:id="@+id/container"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:paddingTop="?attr/actionBarSize"
+    android:background="@color/white">
+
+    <com.google.android.material.bottomnavigation.BottomNavigationView
+        android:id="@+id/nav_view"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="0dp"
+        android:layout_marginEnd="0dp"
+        android:background="?android:attr/windowBackground"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintLeft_toLeftOf="parent"
+        app:layout_constraintRight_toRightOf="parent"
+        app:menu="@menu/bottom_nav_menu" />
+
+    <fragment
+        android:id="@+id/nav_host_fragment"
+        android:name="androidx.navigation.fragment.NavHostFragment"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        app:defaultNavHost="true"
+        app:layout_constraintBottom_toTopOf="@id/nav_view"
+        app:layout_constraintLeft_toLeftOf="parent"
+        app:layout_constraintRight_toRightOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        app:navGraph="@navigation/nav_graph" />
+
+</androidx.constraintlayout.widget.ConstraintLayout>
+```
+A NavHost is an empty container used to display destinations from the navigation graph
+
+**Note : An error for `app:menu="@menu/bottom_nav_menu"` is seen.This is because it does not exist. To solve this, click `Alt`+ `Enter` and select "Create resource file for bottom_nav_menu.xml"**
+
+
+![Create a resource file](/engineering-education/bottom-navigation-bar-in-android/createmenu.jpg)
+
+This menu will pop up.
+
+![Menu](/engineering-education/bottom-navigation-bar-in-android/menu.jpg)
+
+Select *OK* and the error is fixed.
+
+### Step 4 -Creating A Fragment.
+A **Fragment** is a sub-activity. Fragments are used to simplify the reuse of components and logic in different layouts. First, navigate to the java directory and right-click. Select *New*. Click *Fragment* and select *Fragment(Blank)*. Name the Fragment and select *Finish*.
```

Comment:
A good way to show steps in an IDE is to use the arrows. Something like "Select `New -> Fragment -> Fragment(blank)`" can be used.